fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -296,6 +309,87 @@ public DataApiRequest(
     }
 
     /**
+     * To check whether dateTime column request is first one in the sort list or not.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return True if dateTime column is first one in the sort list. False otherwise
+     */
+    protected Boolean isDateTimeFirstSortField(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null) {
+            List<String> columns = new ArrayList(sortColumns.keySet());
+            return columns.get(0).equals(DATE_TIME_STRING);
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * Method to remove the dateTime column from map of columns and its direction.
+     *
+     * @param sortColumns  map of columns and its direction
+     *
+     * @return  Map of columns and its direction without dateTime sort column
+     */
+    protected Map<String, SortDirection> removeDateTimeSortColumn(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null && sortColumns.containsKey(DATE_TIME_STRING)) {
+            sortColumns.remove(DATE_TIME_STRING);
```

Comment:
During the creation of `<OrderByColumn> dateTime`, It is assured that dateTime is first in the list. If not, it will show an exception. If the code reaches this point, it assures dateTime is first in the sort list.

Reason for not considering List is, 
- I need to split each element in a list for validations like 'dateTime' is first in the list
- Need to split to check if it is requested middle of the list. So i can throw valid error to the user
- Split during regular sorting.

Instead of all the different splits, I have done it once and associating with its direction. We can avoid all the other splits unnecessarily.